src/c++/perf_analyzer/infer_data_manager_shm.cc

```diff
@@ -90,108 +99,69 @@ InferDataManagerShm::Init()
         region_name, shared_memory_type_, alloc_size,
         reinterpret_cast<void**>(&output_shm_ptr)));
   }
+  return cb::Error::Success;
+}
 
+cb::Error
+InferDataManagerShm::CreateAndPopulateInputMemoryRegions()
+{
+  // All combinations of input + stream + step
+  //
   for (const auto& input : *(parser_->Inputs())) {
     const std::string& name = input.first;
     const ModelTensor& tensor = input.second;
-    for (int i = 0; i < (int)data_loader_->GetDataStreamsCount(); i++) {
```

Comment:
This looks like a cleanup/refactor. Is that true?

Is `CreateAndPopulateInputMemoryRegion()` holding a lot of the logic? Can you give a brief summary of what changed?

Reply:
Yes, this was a full cut and paste into CreateAndPopulateInputMemoryRegions (and then split into CreateAndPopulateMemoryRegion, and then split into GetInputData and ValidateShapeTensor) to help make sure I fully understood what the shm case did to help me model the non-shm case.

This was all done in my first commit in the branch. No logical changes were made.